src/parser.rs

```diff
@@ -3078,6 +3078,13 @@ impl<'a> Parser<'a> {
     }
 
     pub fn parse_drop(&mut self) -> Result<Statement, ParserError> {
+        // MySQL dialect supports `TEMPORARY`
+        let temporary = if dialect_of!(self is MySqlDialect) {
```

Comment:
Can we also please parse if it is a dialect of `GenericDialect`? Generic Dialect is sort of a catchall dialect that supports everything